Plugins/JiraCommitTipPlugin/JiraComminTipPlugin.cs

```diff
@@ -0,0 +1,126 @@
+        private readonly StringSetting user = new StringSetting("Jira user", string.Empty);
+        private readonly PasswordSetting password = new PasswordSetting("Jira password");
+        private readonly StringSetting filterName = new StringSetting("Filter name", "[Filter should be in your favorites]");
+        private FormCommit commitForm;
+
+        public JiraComminPlugin()
+        {
+            Description = description;
+        }
+
+        public override bool Execute(GitUIBaseEventArgs gitUiCommands)
+        {
+            MessageBox.Show(GetMessageToCommit());
+            return false;
+        }
+
+        public override IEnumerable<ISetting> GetSettings()
+        {
+            yield return url;
+            yield return user;
+            yield return password;
+            yield return filterName;
+        }
+
+        public override void Register(IGitUICommands gitUiCommands)
+        {
+            base.Register(gitUiCommands);
+            gitUiCommands.PostSettings += gitUiCommands_PostSettings;
+            gitUiCommands.PreCommit += gitUiCommands_PreCommit;
+            UpdateJiraSettings();
+        }
+
+        private void UpdateJiraSettings()
+        {
+            jira = new Jira(url[Settings], user[Settings], password[Settings]);
+            try
+            {
+                if (jira == null || jira.GetAccessToken() == null)
+                    return;
+                filter = jira.GetFilters().FirstOrDefault(f => f.Name.Equals(filterName[Settings]));
+            }
+            catch (Exception)
+            {
+                return;
+            }
+        }
+
+        private void gitUiCommands_PostSettings(object sender, GitUIPostActionEventArgs e)
+        {
+            UpdateJiraSettings();
+        }
+
+        public override void Unregister(IGitUICommands gitUiCommands)
+        {
+            base.Unregister(gitUiCommands);
+            gitUiCommands.PreCommit -= gitUiCommands_PreCommit;
+            gitUiCommands.PostSettings -= gitUiCommands_PostSettings;
+        }
+
+        private void gitUiCommands_PreCommit(object sender, GitUIBaseEventArgs e)
+        {
+            commitForm = e.OwnerForm as FormCommit;
+            if (commitForm == null)
+                return;
+            var button = CreateAddInfoBitton();
+            commitForm.AddInfoButton(button);
+        }
+
+        private ToolStripButton CreateAddInfoBitton()
+        {
+            var button = new ToolStripButton
+            {
+                ImageAlign = ContentAlignment.MiddleLeft,
+                Margin = new Padding(1, 3, 3, 3),
+                Name = "AddJiraInfo",
+                Size = new Size(171, 26),
+                Text = description
+            };
+            button.Click -= AddInfo;
+            button.Click += AddInfo;
+            return button;
+        }
+
+        private void AddInfo(object sender, EventArgs e)
+        {
+            if (commitForm == null)
```

Comment:
Sharing the same field for each form isn't very robust. I'm not sure if user can open multiple commit forms from the same app instance..
Why not just send the form to the `CreateAddInfoButton` as argument and capture to subscription?